android/app/src/main/java/app/organicmaps/location/TrackRecordingService.java

```diff
@@ -0,0 +1,172 @@
+package app.organicmaps.location;
+
+import android.app.ForegroundServiceStartNotAllowedException;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.os.Build;
+import android.os.IBinder;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.annotation.RequiresPermission;
+import androidx.core.app.NotificationChannelCompat;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+import androidx.core.content.ContextCompat;
+import app.organicmaps.MwmActivity;
+import app.organicmaps.MwmApplication;
+import app.organicmaps.R;
+import app.organicmaps.settings.TrackRecordSettingsFragment;
+import app.organicmaps.util.LocationUtils;
+import app.organicmaps.util.log.Logger;
+
+import static android.Manifest.permission.ACCESS_FINE_LOCATION;
+
+public class TrackRecordingService extends Service implements LocationListener
+{
+  public static final String TRACK_REC_CHANNEL_ID = "TRACK RECORDING";
+  public static final int TRACK_REC_NOTIFICATION_ID = 54321;
+  private static NotificationCompat.Builder mNotificationBuilder;
+  private static final String TAG = TrackRecordingService.class.getSimpleName();
+
+  @Nullable
+  @Override
+  public IBinder onBind(Intent intent)
+  {
+    return null;
+  }
+  @RequiresPermission(value = ACCESS_FINE_LOCATION)
+  public static void startForegroundService(@NonNull Context context)
```

Comment:
I have transferred the calling of service from preference class to mwmActivity class.

Although without notification permission i don't think it will show any notification. I tried with navigation as well but no notification shows if notification is turned off in settings.

Reply:
This problem is still their even after calling the requestPostNotification() method from MwmActivity, It do not asks permissions for notifications even if it disabled in setiings although functionality works fine but it do not shows any notification. 
Please test this with navigation as well. @rtsisyk